evaluators/base.py

```
@@ -1,54 +1,184 @@
         self.name = name
         self.config = config
+        self.timeout = config.get('timeout', 45)
+        self._available = False
+        
+        logger.info(f"Evaluator initialized: {name}")
     
     @abstractmethod
-    def evaluate_answers(self, questions: List[str], answers: List[str], 
-                        ground_truths: List[str], contexts: List[List[str]] = None) -> Dict[str, List[float]]:
+    async def evaluate_answers_async(self, questions: List[str], answers: List[str], 
+                                  ground_truths: List[str], contexts: List[List[str]] = None) -> Dict[str, List[float]]:
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)